common/scala/src/main/scala/whisk/common/Logging.scala

```diff
@@ -38,50 +35,58 @@ trait Logging {
+    def setVerbosity(level: LogLevel) =
         this.level = level
 
     def getVerbosity() = level
 
     def setComponentName(comp: String) =
         this.componentName = comp
 
-    def debug(from: AnyRef, message: String, marker: LogMarkerToken = null)(implicit id: TransactionId = TransactionId.unknown) = {
-        val mark = id.mark(marker)
-        if (level == Verbosity.Debug || mark.isDefined)
-            emit("DEBUG", id, from, message, mark)
+    def debug(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = {
+        if (level >= DebugLevel) {
+            emit(DebugLevel, id, from, message)
+        }
     }
 
-    def info(from: AnyRef, message: String, marker: LogMarkerToken = null)(implicit id: TransactionId = TransactionId.unknown) = {
-        val mark = id.mark(marker)
-        if (level != Verbosity.Quiet || mark.isDefined)
-            emit("INFO", id, from, message, mark)
+    def info(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = {
+        if (level >= InfoLevel) {
+            emit(InfoLevel, id, from, message)
+        }
     }
 
-    def warn(from: AnyRef, message: String, marker: LogMarkerToken = null)(implicit id: TransactionId = TransactionId.unknown) = {
-        val mark = id.mark(marker)
-        if (level != Verbosity.Quiet || mark.isDefined)
-            emit("WARN", id, from, message, mark)
+    def warn(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = {
+        if (level >= WarningLevel) {
+            emit(WarningLevel, id, from, message)
+        }
+
     }
 
-    def error(from: AnyRef, message: String, marker: LogMarkerToken = null)(implicit id: TransactionId = TransactionId.unknown) = {
-        emit("ERROR", id, from, message, id.mark(marker))
+    def error(from: AnyRef, message: String)(implicit id: TransactionId = TransactionId.unknown) = {
+        if (level >= ErrorLevel) {
+            emit(ErrorLevel, id, from, message)
+        }
     }
 
-    def emit(category: AnyRef, id: TransactionId, from: AnyRef, message: String, mark: Option[LogMarker] = None) = {
-        val now = mark map { _.now } getOrElse Instant.now(Clock.systemUTC)
+    def emit(loglevel: LogLevel, id: TransactionId, from: AnyRef, message: String, marker: Option[LogMarker] = None) = {
+        val now = Instant.now(Clock.systemUTC)
         val time = Logging.timeFormat.format(now)
         val name = if (from.isInstanceOf[String]) from else Logging.getCleanSimpleClassName(from.getClass)
-        val msg = mark map { m => s"[marker:${m.token}:${m.delta}] $message" } getOrElse message
 
-        if (componentName != "") {
-            outputStream.println(s"[$time] [$category] [$id] [$componentName] [$name] $msg")
-        } else {
-            outputStream.println(s"[$time] [$category] [$id] [$name] $msg")
+        val component = Option(componentName).filter(_.trim.nonEmpty).map("[" + _ + "]")
```

Comment:
This will eval to `None` and print so below; instead turn this into a Seq(String) vs Seq() for the mkString to work as expected.